url8012c8d031_dmbb966_BBMS_tgz-pJ8-testbed_JNEATtestJ8/src/unit/JNEATIntegration.java

```diff
@@ -0,0 +1,747 @@
+    		break;
+    	case 2:
+    		GlobalFuncs.defaultOrgType = OrganismTypeEnum.SIMPLE_DUAL;
+    		break;
+    	case 4:
+    		GlobalFuncs.defaultOrgType = OrganismTypeEnum.BASE_MAXHEX;
+    		break;
+    	case 3:
+    		GlobalFuncs.defaultOrgType = OrganismTypeEnum.BASE_RANDOM;
+    		break;
+    	case 7:
+    		GlobalFuncs.defaultOrgType = OrganismTypeEnum.SIX_DIRECTIONAL;
+    		break;
+		default:
+			GlobalFuncs.defaultOrgType = OrganismTypeEnum.SIMPLE_SINGLE;
+			GUI_NB.GCO("ERROR!  Could not determine organism type based on population file.  Defaulting to simple single.");
+			break;    			
+    	}
+		
+		GlobalFuncs.newEpoch = true;				
+		
+		GlobalFuncs.COAIndex = 1;
+		JNEATIntegration.TestIterationSetup(numScouts);
+		
+		// GUI_NB.GCO("Now initiating stuff.");
+		//TestIterationSetup(numScouts);		
+    }
+    
+    public static void PopTestNewEpoch(int newEpoch) {
+    	GlobalFuncs.currentPopEpochNum = newEpoch;
+    	GlobalFuncs.targetPop = FIO.PoppyCock(GlobalFuncs.currentPopEpochNum);
+    	GlobalFuncs.currentPop = new Population(GlobalFuncs.targetPop);
+    	GUI_NB.GCO("Population data read.  Initializing scenario with " + GlobalFuncs.currentPop.organisms.size() + " orgs");
+    	
+    	GlobalFuncs.newEpoch = true;
+    	GlobalFuncs.COAIndex = 1;
+    	JNEATIntegration.TestIterationSetup(GlobalFuncs.numScoutsPer);
+    	
+    	ClockControl.SetPaused(false);
+    }
+    
+
+	
+	public static void ScenIterationFromFile() {
+		String fullPath = "src/saves/" + GlobalFuncs.tempStr;
+		File popFile = FIO.newFile(fullPath);
+		if (!popFile.exists()) {
+			GUI_NB.GCO("Error reading population file!");
+			
+			DialogLoadScen x = new DialogLoadScen(GlobalFuncs.gui, true);
+			x.setVisible(true);
+		}
+		else {
+			
+			GUI_NB.GCO("Loading population data.");
+			Path p = popFile.toPath();
+			GlobalFuncs.currentPop = new Population(p);
+			GUI_NB.GCO("Population data read.  Initializing scenario with " + GlobalFuncs.currentPop.organisms.size() + " orgs");
+			GlobalFuncs.newEpoch = true;
+			
+			int numScouts = GlobalFuncs.currentPop.PopulationSlice(GlobalFuncs.percentPerRun);
+			
+			JNEATIntegration.ScenIterationSetup(numScouts);
+		}
+	}
+	
+	
+	public static void TestIterationSetup(int numScouts) {
+		GlobalFuncs.allSpots.records.clear();
+		GlobalFuncs.numScoutsPer = numScouts;
+		
+		// Sequentially choose a COA
+		GlobalFuncs.curCOA = GlobalFuncs.allCOAs.elementAt(GlobalFuncs.COAIndex - 1);
+		GlobalFuncs.curCOA.LoadCOA();
+		
+		if (numScouts == 0) return;
+		
+		GUI_NB.GCO("New test iteration: Populating " + numScouts + " scouts of pop size " + GlobalFuncs.currentPop.organisms.size());
+		
+		// First, eliminate friendly units from the unit roster
+		for (int i = 0; i < GlobalFuncs.friendlyUnitList.size(); i++) {
+			Unit finger = GlobalFuncs.friendlyUnitList.elementAt(i);
+			if (finger.emplaced) finger.RemoveSharedSpotting(); 
+			finger.location.HexUnit = null;
+			GlobalFuncs.unitList.remove(finger);					
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 638:643, 695:700.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.